src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+import com.epam.esm.model.entity.Page;
+import com.epam.esm.model.entity.QuerySpecification;
+import com.epam.esm.model.exception.CertificateNotFoundException;
+import com.epam.esm.model.exception.OrderNotFoundException;
+import com.epam.esm.service.CertificateService;
+import com.epam.esm.util.MapperDTO;
+import lombok.AllArgsConstructor;
+import org.apache.commons.lang3.ObjectUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class CertificateServiceImpl implements CertificateService {
+
+    private final CertificateDAO certificateDAO;
+    private final TagDAO tagDAO;
+    private final OrderDAO orderDAO;
+    private final MapperDTO mapperDTO;
+
+    @Override
+    @Transactional
+    public CertificateDTO create(CertificateDTO certificateDTO) {
+        Certificate certificate = mapperDTO.convertDTOToCertificate(certificateDTO);
+        certificate.setTags(new HashSet<>());
+        attachTags(certificate, certificateDTO.getTags());
+        certificate = certificateDAO.create(certificate);
+        return mapperDTO.convertCertificateToDTO(certificate);
+    }
+
+    @Override
+    public List<CertificateDTO> findAll(Page page) {
+        return certificateDAO.findAll(page)
+            .stream()
+            .map(mapperDTO::convertCertificateToDTO)
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    public CertificateDTO findById(Long id) {
+        Certificate certificate = certificateDAO.findById(id)
+            .orElseThrow(() -> new CertificateNotFoundException(id.toString()));
+        if (!certificate.isActive()) {
+            throw new CertificateNotFoundException(id.toString());
+        }
+        return mapperDTO.convertCertificateToDTO(certificate);
+    }
+
+    @Override
+    public List<CertificateDTO> findAll(ParametersSpecificationDTO querySpecificationDTO,
+                                        Page page) {
+        QuerySpecification querySpecification = mapperDTO.convertDTOToQuery(querySpecificationDTO);
+        return certificateDAO.findAll(querySpecification, page)
+            .stream()
+            .map(mapperDTO::convertCertificateToDTO)
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    @Transactional
+    public CertificateDTO update(CertificateDTO updateDTO) {
+        Optional<Certificate> optional = certificateDAO.findById(updateDTO.getId());
+        if (optional.isEmpty() || !optional.get().isActive()) {
+            throw new CertificateNotFoundException(updateDTO.getId().toString());
+        }
+        Certificate certificate = optional.get();
+        Certificate update = mapperDTO.convertDTOToCertificate(updateDTO);
+        certificate = certificateDAO.update(certificate, update);
```

Comment:
update logic should be in the service layer, please move certificateDAO.update(certificate, update) from DAO layer to service